lib/core/src/stringOpr.cpp

```diff
@@ -45,28 +48,28 @@ char *rmemcpy( void *dest, const void *src, int strLen, int maxLen ) {
     }
 }
 
-char *rstrcpy( char *dest, const char *src, int maxLen ) {
+char* rstrcpy(char* dest, const char* src, int maxLen)
+{
     // snprintf with logging on fail
-
-    if ( dest == NULL || src == NULL ) {
-        return NULL;
+    if (dest == nullptr || src == nullptr) {
+        return nullptr;
     }
-    int status = snprintf( dest, maxLen, "%s", src );
 
-    if ( status >= 0 && status < maxLen ) {
-        return dest;
-    }
-    else if ( status >= 0 ) {
-        rodsLog( LOG_ERROR,
-                 "rstrcpy not enough space in dest, slen:%d, maxLen:%d, stacktrace:\n%s",
-                 status, maxLen, irods::stacktrace().dump().c_str());
-        rodsLog( LOG_DEBUG, "rstrcpy arguments dest [%s] src [%s]", dest, src );
-        return NULL;
-    }
-    else {
-        rodsLog( LOG_ERROR, "rstrcpy encountered an encoding error." );
-        return NULL;
+    int status{std::snprintf(dest, maxLen, "%s", src)};
+
+    if (status >= 0) {
+        if (status < maxLen) {
+            return dest;
+        }
+        log_server::error(
+            "rstrcpy not enough space in destination buffer, source length:{}, destination length:{}; "
+            "resulting state of destination [{}], truncated from source [{}]",
+            status, maxLen, dest, src);
+        return nullptr;
     }
+
+    log_server::error("rstrcpy encountered an encoding error.");
```

Comment:
Is this always the case? Maybe we could show the `errno` set by the system in this case (https://pubs.opengroup.org/onlinepubs/9699919799/functions/fprintf.html#tag_16_159_04).

I'm also okay with leaving this the way that it is because I don't like this function anyway.